src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -140,19 +137,7 @@ internal byte[] DecryptCek(X509Certificate2 cert, RSA privateKey, out Exception
                     recipient.RecipientIdentifierType.ToString());
             }
 
-            RSAEncryptionPadding padding = recipient.RSAEncryptionPadding;
-
-            if (padding is null)
-            {
-                if (recipient.Certificate.GetKeyAlgorithm() == Oids.RsaOaep)
-                {
-                    padding = RSAEncryptionPadding.OaepSHA1;
-                }
-                else
-                {
-                    padding = RSAEncryptionPadding.Pkcs1;
-                }
-            }
+            RSAEncryptionPadding padding = recipient.RSAEncryptionPadding ?? RSAEncryptionPadding.Pkcs1;
```

Comment:
Since the X509Pal doesn't support RSAES_OAEP certificates we can not currently use it in the managed PAL, so there is no point in checking if the key algorithm is id `id-RSAES-OAEP`.